int32 rolled
int32 threshold
int32 numSides
int32 type
